src/Build.UnitTests/ProjectCache/ProjectCacheTests.cs

```diff
@@ -1475,6 +1499,90 @@ public void ParallelStressTest(bool useSynchronousLogging, bool disableInprocNod
             cache.QueryStartStops.Count.ShouldBe(graph.ProjectNodes.Count * 2);
         }
 
+        [Fact]
+        public void ProxyCacheHitsOnPreviousCacheMissesShouldWork()
```

Comment:
A few comments in this method would be nice :) The important part is that you schedule things that hit and things that don't in the same request, with the "don't" case first?